src/commonMain/kotlin/sim/SimParticipant.kt

```diff
@@ -22,7 +22,7 @@ class SimParticipant(val character: Character, val rotation: Rotation, val sim:
     val logger = KotlinLogging.logger {}
 
     var stats: Stats = Stats()
-    lateinit var resource: Resource
+    lateinit var resource: MutableMap<Resource.Type, Resource>
```

Comment:
I think this type can be just `Map` - the object does need to be initially mutable to be created below, but the property can be the more generic non-mutable type.

Reply:
Done.